### ISSUE TITLE
Improve support for engines

### ISSUE THREAD

**Issue by cupakromer** [rspec/rspec-rails#1227]
This is probably going to be a relatively large task. We've recently had a [few](https://github.com/rspec/rspec-rails/issues/1203) [issues](https://github.com/rspec/rspec-rails/issues/1226) show up regarding needing improved support for testing engine parts in isolation. While those issues are provided for via a relatively simple module inclusion, general engine support is lacking.

I started to follow the newish [Rails Guide: Getting Started with Engines](http://edgeguides.rubyonrails.org/engines.html). I followed the tutorial using today's latest stable Rails release (4.1.8). Essentially, right from the beginning RSpec is lagging very far behind :frowning:. Attempting to run the scaffold generator, `bin/rails generate scaffold article title:string text:text`, resulted in zero RSpec generated files. :weary: Also, if you do not run the default test-unit process when creating the engine there isn't a clear way on generating another dummy test application.

I'm happy to begin working on making engine author's lives easier. I'm also happy to take patches and PRs / suggestions / advise from the community.

Cheers everyone! :smile_cat:

**Comment by mikeastock** [comment]
@cupakromer do you have any ideas on where to start?

**Comment by cupakromer** [comment]
@mikeastock not exactly. I actually need to create an engine based gem at my work. I'm going to use that as a driving force for improving this. I guess I'll just be starting from the basics and trying to get everything setup smoothly. Since I haven't done too much with engines in the past, I'm sure I'll hit many of the standard pain points right out of the gate.

**Comment by enricostano** [comment]
cool @mikeastock !

I would be glad to help you on this, just ping me if you need it. :smiley:
